pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -417,6 +417,18 @@ func (c *Cluster) startOSDDaemonsOnPVC(pvcName string, config *provisionConfig,
 	// start osds
 	for _, osd := range osds {
 		logger.Debugf("start osd %v", osd)
+
+		// keyring must be generated before deployment creation in order to avoid a race condition resulting
```

Comment:
Since this code block is called from two different methods, i'd suggest factoring out to a helper method so there is less duplicate code.